Scripts/Commands: Fix guid usage and enhance commands

Rochet2 opened the pull request:
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

- Fix targeting gameobjects and creatures for .dist and .gps. The old code used DB guid to search from core generated guid lists.
- Fix some error messages telling no player found even if creatures and gameobjects and players were searched
- Change atoi to atoul when receiving lowguids as string or other uint32 values
- Use ObjectGuid::LowType instead of uint32 for lowguid type
- Allow .gob info to take in gameobject guid link as well as entry link in addition to gameobject entry and update the documentation of the command
- Change .gob delete to post the spawnid (dbguid) instead of the ingame generated guid (which is basically never seen or used by the player in any other commands that I see)
- Allow spawntimesecs to be negative when spawning a gameobject
- Disable searching by ingame spawn id in commands. (it makes little sense for a user to provide a number which is then attempted to be used as DB guid and non DB guid when both could have a match)
- Update .gob turn documentation to tell about possibility to provide orientation in command
- Correct the types in .gob add command result trinity_string
- Correct the creature search for .npc del <lowguid>
- Change .event <id> to .event info <id>
- Add .go offset command
- Remove the .go command for empty string (no subcommand given) because it just used .go xyz
- Extend .gob info to show size, faction, flags and model dimensions
- Remove some unnecessary casts
- Document .wp show better (first and last were not documented)
- Insert spawnid to .wp show info error message
- Fix spawntime printing for .gobject target
- Fix guid targetting for .npc set movetype <guid> <type>
- Fix query by spawnid for .wp show info
- Fix deleting of existing waypoints when twice doing the command .wp show on
- Fix deleting of existing waypoints when doing the command .wp show off
- Change wpguid column in DB and in core to uint32, which is what the spawnid type is in core and db.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)
Closes https://github.com/TrinityCore/TrinityCore/issues/18241
Closes https://github.com/TrinityCore/TrinityCore/issues/16733

**Tests performed:** (Does it build, tested in-game, etc.)

- [x] Does it build?
- [x] Tested ingame? Most important parts tested. For more see TODO

**Known issues and TODO list:** (add/remove lines as needed)
The PR was made for 3.3.5, but probably also applies to master branch as well.
**NOTE** when converting to 6.x+ you should change atoul to atoull because ObjectGuid::LowType is defined as uint32 on 3.3.5 and uint64 on 6.x+

Test affected commands:
- [x] .damage
- [x] .dist
- [x] .gps
- [x] .gob set state
- [x] .gob info
- [x] .gob set phase
- [x] .gob move
- [x] .gob turn
- [x] .gob delete
- [x] .gob add temp
- [x] .gob add
- [x] .gob activate
- [x] .go obj
- [x] .go crea
- [x] .list object
- [x] .npc del <guid>
- [x] .achievement add
- [x] .event info
- [x] .event start
- [x] .event stop
- [x] .go creature
- [x] .go graveyard
- [x] .go grid
- [x] .go taxinode
- [x] .go trigger
- [x] .go zonexyz
- [x] .go xyz
- [x] .go ticket
- [x] .npc add
- [x] .npc delete
- [x] .go
- [x] .go offset
- [x] .gob info
- [x] .event
- [x] .wp show on
- [x] .wp show off
- [x] .npc set movetype
- [x] .gob tar
- [x] .wp show info

Rochet2 committed:
Disable search by ingame generated guid

Rochet2 committed:
Use atoul and ObjectGuid::LowType for guids in commands

Rochet2 committed:
Fix creature and gameobject links in .gps and .dist commands

Rochet2 committed:
Allow .gob info to use entry or shift clicked link

Rochet2 committed:
Fix documentation and trinity_string for some existing commands

Rochet2 commented:
It seems .npc delete for example does not work as intended either.
So if this is merged now I will make a new PR for them then, otherwise will add them here.

Rochet2 committed:
Change atoi to atoul in many places. Fix .npc del <lowguid>

Rochet2 committed:
Change .event <id> to .event info <id>

Rochet2 commented:
Guess thats all. review and advice with SQLs needed.

Keader commented:
```The command "mysql -utrinity -ptrinity auth < sql/base/auth_database.sql" exited with 0.
0.01s$ ./contrib/check_updates.sh auth 3.3.5 auth
Database Updater check script:
  Checking database 'auth' for missing filenames in tables...
- "sql/updates/auth/2016_11_19_xx_auth.sql" is missing in the 'auth'.'updates' table.
Fatal error:
  The Database Updater is broken for the 'auth' database,
  because the applied update is missing in the 'auth'.'updates' table.
How to fix:
  Insert the missing names of the already applied sql updates
  to the 'updates' table of the 'auth' base dump ('sql/base/auth_database.sql').
The command "./contrib/check_updates.sh auth 3.3.5 auth" exited with 1.
0.45s$ mysql -utrinity -ptrinity characters < sql/base/characters_database.sql
```

Rochet2 commented:
@Keader But if the name and maybe contents of the SQL change, then that base sql must be edited again, no?

Aokromes commented:
yeah, that thing must be added by the person merging the pr.

Rochet2 committed:
Extend .gob info command

Rochet2 committed:
Add .go offset command

P-Kito commented on src/server/scripts/Commands/cs_go.cpp:
```diff
@@ -492,7 +492,7 @@ class go_commandscript : public CommandScript
         float z;
         float ort = port ? (float)atof(port) : player->GetOrientation();
-        uint32 mapId = id ? (uint32)atoi(id) : player->GetMapId();
+        uint32 mapId = id ? (uint32)atoul(id) : player->GetMapId();
```
Is this cast to uint32 needed?

Keader commented:
> - "sql/updates/auth/2016_11_19_xx_auth.sql" is missing in the 'auth'.'updates' table.
> Fatal error:
>   The Database Updater is broken for the 'auth' database,
>   because the applied update is missing in the 'auth'.'updates' table.
> How to fix:
>   Insert the missing names of the already applied sql updates
>   to the 'updates' table of the 'auth' base dump ('sql/base/auth_database.sql').

Aokromes commented:
https://github.com/TrinityCore/TrinityCore/pull/18278#issuecomment-261768207

Rochet2 committed:
Remove casts

Rochet2 committed:
Rename SQL files to xxxx for easier finding and editing later on

Rochet2 committed:
Fix .wp .gob tar .npc set movetype

Remove unnecessary casts
Change uint32 parses from atoi to atoul
Fix spawntime printing for .gobject target
Fix guid targetting for .npc set movetype <guid> <type>
Fix query by spawnid for .wp show info
Fix deleting of existing waypoints when twice doing the command .wp show on
Fix deleting of existing waypoints when doing the command .wp show off
Change wpguid column in DB and in core to uint32, which is what the spawnid type is in core and db.

Rochet2 committed:
Document .wp show better

Rochet2 committed:
Insert spawnid to .wp show info error message

Keader commented:
why this world files (sql) is not together?

Rochet2 commented:
No reason, should they be?

Rochet2 committed:
Rename and merge sql files, fix typo in 'slected_waypoint'

Keader commented:
y, all world changes in same file please :D

Rochet2 commented:
well, did that.

Commit:
Update auth_database.sql

Add A68511D0BF94C41F720A11B630CCB36BBEA6B17D
and 2017_01_08_00_auth.sql to the `auth`.`updates` table.

Commit:
Rename xxxx_xx_xx_xx_auth.sql to 2017_01_08_00_auth.sql

Commit:
Rename xxxx_xx_xx_xx_world.sql to 2017_01_08_00_world.sql
